frontend/src/components/assessment-creation/FilterMenu.tsx

```diff
@@ -1,12 +1,130 @@
-import React from "react";
-import { Button } from "@chakra-ui/react";
+import React, { SetStateAction } from "react";
+import {
+  Button,
+  Text,
+  Popover,
+  PopoverBody,
+  PopoverContent,
+  PopoverFooter,
+  PopoverTrigger,
+  Divider,
+  Flex,
+} from "@chakra-ui/react";
+import { Select } from "chakra-react-select";
 import { FunnelIcon } from "../../assets/icons";
+import { MultiOption } from "../../types/SelectInputTypes";
 
-const FilterMenu = (): React.ReactElement => {
+export type FilterMenuProp = {
+  label: string;
+  setState: React.Dispatch<SetStateAction<Array<string>>>;
+  options: MultiOption[];
+};
+
+interface FilterMenuProps {
+  filterProps: FilterMenuProp[];
+}
+
+function addFilter(
+  optionsSelected: Map<string, string[]>,
+  filterProps: FilterMenuProp[],
+): void {
+  filterProps.forEach((prop: FilterMenuProp) => {
+    if (optionsSelected.has(prop.label) && optionsSelected.get(prop.label)) {
+      prop.setState(optionsSelected.get(prop.label) as string[]);
+    } else {
+      prop.setState([]);
+    }
+  });
+}
+
+const FilterMenu = ({ filterProps }: FilterMenuProps): React.ReactElement => {
+  const [optionsSelected, setOptionsSelected] = React.useState<
+    Map<string, string[]>
+  >(new Map());
+
+  const attributeList = filterProps.map((filterProp, i) => (
+    <>
+      <Flex pt="2%" px={3} pb={4} justifyContent="space-around">
+        <div style={{ width: "7rem" }}>
+          <Text color="blue.300" textStyle="link">
+            {filterProp.label}
+          </Text>
+        </div>
+
+        <div style={{ width: "23rem" }}>
+          <Select
+            chakraStyles={{
+              dropdownIndicator: (provided) => ({
+                ...provided,
+                background: "white",
+              }),
+              downChevron: (provided) => ({
+                ...provided,
+                boxSize: "1.5rem",
+              }),
+              crossIcon: (provided) => ({
+                ...provided,
+                boxSize: "0.8rem",
+              }),
+            }}
```

Comment:
can we move styling for `<Select>` into the `themes` folder?

Reply:
So for Select, it's not a native Chakra component, but rather a wrapper component imported from `chakra-react-select`, and so the default theme doesn't work on it from what I'm looking into. In the [chakra-react-select docs](https://www.npmjs.com/package/chakra-react-select#theme-styles), it describes how individual components can be altered, but it says to use chakrastyles if you don't want to affect all components. 

"Only make changes to your global component themes if you want them to appear in all instances of that component. Otherwise, just change the individual components' styles using the chakraStyles prop."

I'm not how to approach the styling, but I'd love to hear thoughts!